src/tests/verify_resources.py

```diff
@@ -0,0 +1,196 @@
+import collections
+import glob
+import yaml
+import sys
+import os
+import re
+
+_topdir = os.path.abspath(os.path.dirname(__file__) + "/../..")
+
+# autodict is a defaultdict returning a new autodict for missing keys.
+# __add__ allows `d[k] += v` to work automatically before `d` contains `k`.
+class autodict(collections.defaultdict):
+    def __init__(self,*other):
+        collections.defaultdict.__init__(self, self.__class__, *other)
+    def __add__ (self, other):
+        return other
+    def __repr__(self):
+        return '%s(%s)' % (self.__class__.__name__, dict.__repr__(self))
+
+def rgfilter(fn):
+    return not (fn.endswith("_downtime.yaml") or fn.endswith("/SITE.yaml"))
+
+def rgname(fn):
+    return re.search(r'/([^/]+)\.yaml$', fn).group(1)
+
+def sumvals(d):
+    return sum(d.values())
+
+# temporary list of old vo names, until they are purged from our data
+def get_disabled_vo_names():
+    return set(['CompBioGrid', 'Engage', 'GROW', 'LBNL_DSD', 'NEBioGrid',
+                'NWICG', 'NYSGRID', 'OSGEDU', 'UC3', 'superbvo.org'])
+
+
+def get_vo_names():
+    return set( re.search(r'/([^/]+)\.yaml$', path).group(1) for path in
+                glob.glob(_topdir + "/virtual-organizations/*.yaml") )
+
+def load_yamlfile(fn):
+    with open(fn) as f:
+        return yaml.safe_load(f)
+
+def main():
+    global services
+    os.chdir(_topdir + "/topology")
+
+    yamls = sorted(glob.glob("*/*/*.yaml"))
+    rgfns = list(filter(rgfilter, yamls))
+    rgs = list(map(load_yamlfile, rgfns))
+    #facility_site_rg = [ fn[:-len(".yaml")].split('/') for fn in rgfns ]
+
+    errors  = test_1_rg_unique(rgs, rgfns)
+    errors += test_2_res_unique(rgs, rgfns)
+    errors += test_3_voownership(rgs, rgfns)
+    errors += test_4_res_svcs(rgs, rgfns)
+    errors += test_5_sc(rgs, rgfns)
+    errors += test_6_site()
+
+    print("%d Resource Group files processed." % len(rgs))
+    if errors:
+        print("%d error(s) encountered." % errors)
+        return 1
+    else:
+        print("A-OK.")
+        return 0
+
+
+def test_1_rg_unique(rgs, rgfns):
+    # 1. Name (file name) of RG must be unique across all sites
+
+    errors = 0
+    rgmap = autodict()
+
+    for rgfile in rgfns:
+        rgmap[rgname(rgfile)] += [rgfile]
+
+    for name, rgflist in sorted(rgmap.items()):
+        if len(rgflist) > 1:
+            print("Resource Group '%s' mentioned for multiple Sites:" % name)
+            for rgfile in rgflist:
+                print(" - %s" % rgfile)
+            errors += len(rgflist) - 1
+
+    return errors
+
+
+def test_2_res_unique(rgs, rgfns):
+    # 2. Name of each resource must be present and
+    #    unique across all resources in all sites
+
+    errors = 0
+    r2rg = autodict()
+
+    for rg,rgfn in zip(rgs,rgfns):
+        for r in rg['Resources']:
+            r2rg[r] += [rgfn]
+
+    for r, rgflist in sorted(r2rg.items()):
+        if len(rgflist) > 1:
+            print("Resource '%s' mentioned for multiple groups:" % r)
```

Comment:
Can we print a message above this that says resource names must be unique across the OSG topology?